var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -763,3 +845,114 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+
+        gpu_map = {
+            "35": "K40",
+            "37": "K80",
+            "60": "P100",
+            "70": "V100",
+            "80": "A100",
+            "gfx906": "Mi50",
+            "gfx908": "Mi100",
+            "gfx90a": "Mi250",
+            "gfx90a:xnack-": "Mi250",
+            "gfx90a:xnack+": "Mi250",
+        }
+
+        if "+cuda" in spec:
+            if (len(spec.variants["cuda_arch"].value) > 1) or spec.satisfies("cuda_arch=none"):
+                raise InstallError("CP2K supports only one cuda_arch at a time.")
+            else:
+                gpu_ver = gpu_map[spec.variants["cuda_arch"].value[0]]
+                args += ["-DCP2K_USE_ACCEL=CUDA"]
+                args += [self.define("CP2K_WITH_GPU", gpu_ver)]
+
+        if "+rocm" in spec:
+            if len(spec.variants["amdgpu_target"].value) > 1:
+                raise InstallError("CP2K supports only one amdgpu_target at a time")
+            else:
+                gpu_ver = gpu_map[spec.variants["amdgpu_target"].value[0]]
+                args += ["-DCP2K_USE_ACCEL=HIP"]
+                args += [self.define("CP2K_WITH_GPU", gpu_ver)]
+
+        args += [
+            self.define_from_variant("CP2K_ENABLE_REGTESTS", "enable_regtests"),
+            self.define_from_variant("CP2K_USE_ELPA", "elpa"),
+            self.define_from_variant("CP2K_USE_LIBINT2", "libint"),
+            self.define_from_variant("CP2K_USE_SIRIUS", "sirius"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_COSMA", "cosma"),
+            self.define_from_variant("CP2K_USE_LIBXC", "libxc"),
+            self.define_from_variant("CP2K_USE_LIBTORCH", "pytorch"),
+            self.define_from_variant("CP2K_USE_METIS", "pexsi"),
+            self.define_from_variant("CP2K_USE_SUPERLU", "pexsi"),
+            self.define_from_variant("CP2K_USE_PLUMED", "plumed"),
+            self.define_from_variant("CP2K_USE_SPGLIB", "spglib"),
+            self.define_from_variant("CP2K_USE_VORI", "libvori"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_QUIP", "quip"),
+        ]
+
+        # we force the use elpa openmp threading support. might need to be revisited though
+        args += [
+            self.define(
+                "CP2K_ENABLE_ELPA_OPENMP_SUPPORT",
+                (spec.satisfies("+openmp") and spec.satisfies("+elpa"))
+                or spec.satisfies("^elpa+openmp"),
```

Comment:
This is just an example where readability might improve using the `in` operator
```suggestion
                ("+elpa +openmp" in spec) or ("^elpa +openmp" in spec),
```